### PR TITLE
Enable tenanted paths for opbs cookies

### PR DESCRIPTION
**Pull request by ThaminduR**
### Proposed changes in this pull request

Fix for https://github.com/wso2/product-is/issues/11381

### CODE REVIEW

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java [hunk 1]
```diff
@@ -44,4 +44,12 @@
      */
     Cookie addOPBrowserStateCookie(HttpServletResponse response);
 
+    /**
+     * Adds the browser state cookie with tenant qualified path to the response.
+     *
+     * @param response
+     * @return Cookie
+     */
+    Cookie addOPBrowserStateCookie(HttpServletResponse response, String tenantDomain);
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java:53]: have it with default access modifier

Comment (ThaminduR) [reply components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java]: b6b6a12

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java [hunk 2]
```diff
@@ -44,4 +51,25 @@
...
+    /**
+     * Adds the browser state cookie with tenant qualified path to the response.
+     *
+     * @param response
+     * @return Cookie
+     */
+    default Cookie addOPBrowserStateCookie(HttpServletResponse response, String tenantDomain) {
+
+        ServletCookie
+                cookie = new ServletCookie(OIDCSessionConstants.OPBS_COOKIE_ID, UUID.randomUUID().toString());
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java:63]: Move the implementation logic to DefaultOIDCSessionStateManager.java

**Review by piraveena — CHANGES_REQUESTED** [review]

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java [hunk 3]
```diff
@@ -44,4 +44,14 @@
...
     Cookie addOPBrowserStateCookie(HttpServletResponse response);
 
+    /**
+     * Adds the browser state cookie with tenant qualified path to the response.
+     *
+     * @param response
+     * @return Cookie
+     */
+    default Cookie addOPBrowserStateCookie(HttpServletResponse response, String tenantDomain) {
+        return null;
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java:54]: In default, we can call the old implementation addOPBrowserStateCookie(request)

Comment (ThaminduR) [reply components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateManager.java]: Added in c0887be

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/DefaultOIDCSessionStateManager.java [hunk 4]
```diff
@@ -81,6 +83,27 @@ public Cookie addOPBrowserStateCookie(HttpServletResponse response) {
...
     }
 
+    /**
+     * Adds the browser state cookie with tenant qualified path to the response.
+     *
+     * @param response
+     * @return Cookie
+     */
+    @Override
+    public Cookie addOPBrowserStateCookie(HttpServletResponse response, String tenantDomain) {
```
[the remark this answers is not in the document]
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/DefaultOIDCSessionStateManager.java:93]: Deprecated in c0887be

**Review by senthalan — APPROVED** [review]

**Review by piraveena — APPROVED** [review]

**Pull request merged**